### DISCUSSION TITLE
Release Pipeline

### DISCUSSION THREAD

**Discussion by badcel** [DevToys-app/DevToys#1250] [General]
Are there any plan to open up the release pipeline of DevToys 2.0?

For me it is unexpected to not be able to verify how the binaries for an open source project are build.

If the build would be integrated with GitHub it would be even better as it would make it transparent how the code flows through the build process into a released binary.

**Comment by veler** [comment]
Hello,

It would indeed help contributors to understand how the code flows through the build process. But I'm currently not planning to open source the pipeline.

First, our pipeline is (at my taste) awfully complex.

Second, this complexity is actually a benefit for us: it makes it hard for outsiders to easily make their own release build identical to ours and publish it, because it's currently hard to reverse engineer it without seeing it. It's indirectly a way for us to protect ourselves from people who might be tempted to take our source code, change the name and logo, remove the license (and don't credit us), and **_sell_** the software. While our MIT license technically permits it, it would be very frustrating to see it happening. We also failed at finding a very permissive license that would prevent what I mentioned above, and we didn't want to prevent the use of DevToys in a commercial purpose (understand, companies using DevToys as part of their employee's workflow), which is why we kept MIT.

It unfortunately happened to another amazing FOSS project. See these 2 links:
https://mathewsachin.github.io/blog/2023/04/09/captura-unmaintained.html
https://github.com/MathewSachin/Captura/issues/405
